src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -2123,8 +2124,9 @@ public final void clearBatch() throws SQLServerException {
                 try (SQLServerStatement stmt = (SQLServerStatement) connection.createStatement(
                         ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY, connection.getHoldability(),
                         stmtColumnEncriptionSetting);
-                        SQLServerResultSet rs = stmt.executeQueryInternal(
-                                "sp_executesql N'SET FMTONLY ON SELECT * FROM " + tableName + " '");) {
+                        SQLServerResultSet rs = stmt
```

Comment:
rs not closed

Reply:
Its inside try-w-resource try block - hence does get closed

Reply:
oops missed that